src/pages/masternodes.tsx

```diff
@@ -0,0 +1,104 @@
+
+import { useEffect, useState } from 'react'
+import { NextPage } from 'next'
+import { useWhaleApiClient } from '../layouts/contexts/WhaleContext'
+
+import { masternodes } from '@defichain/whale-api-client'
+import Loader from '../components/shared/Loader'
+
+const MasternodeItem = ({ data }): JSX.Element => {
+  return (
+    <div data-testid='masternode_item' className='flex flex-col md:flex-row justify-center md:justify-between border-b border-gray-300'>
+      <div className='p-4 text-center md:text-left  md:w-1/4 break-all'>{data.owner.address}</div>
+      <div className='p-4 text-center md:text-right md:w-1/4 break-all'>{data.operator.address}</div>
+      <div className='p-4 text-center md:text-right'>{data.creation.height}</div>
+      <div className='p-4 text-center md:text-right'>{data.resign.height}</div>
+      <div className='p-4 text-center md:text-right'>{data.mintedBlocks}</div>
+      <div className='p-4 text-center md:text-right'>{data.state}</div>
+    </div>
+  )
+}
+
+const Masternodes: NextPage = (): JSX.Element => {
+  const [hasNext, setHasNext] = useState<boolean>(false)
+  const [nextToken, setNextToken] = useState<string | number | undefined>(undefined)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [masternodes, setMasternodes] = useState<masternodes.MasternodeData[]>([])
+
+  // whaleAPI hook
+  const api = useWhaleApiClient()
+
+  useEffect(() => {
+    const fetchMasternodes = async (): Promise<void> => {
+      // NOTE(siradji) Made 2 the default query size. Will be changing it when design is finalized.
+      const response = await api.masternodes.list(2)
+      setLoading(false)
+      setMasternodes(response as any)
+      setHasNext(response.hasNext)
+      setNextToken(response.nextToken)
+    }
+
+    if (loading) {
+      void fetchMasternodes()
+    }
+  }, [])
+
+  // TODO(siradji) Figure out a way to  implement previous button.
+  const handlePreviousPagination = async (): Promise<void> => {}
```

Comment:
use "lazy loading" or "load next" instead of pagination, number-based loading is incredibly harder to scale

Reply:
Yeah i was also thinking of infinite scroll. I will do that.